src/qibolab/transpilers/placer.py

```diff
@@ -0,0 +1,261 @@
+import random
+
+import networkx as nx
+from qibo.config import log, raise_error
+from qibo.models import Circuit
+
+from qibolab.transpilers.abstract import Placer, create_circuit_repr
+
+
+def check_placement(circuit: Circuit, layout: dict, verbose=False) -> bool:
+    """Checks if layout is correct and matches the number of qubits of the circuit.
+
+    Args:
+        circuit (qibo.models.Circuit): Circuit model to check.
+        layout (dict): physical to logical qubit mapping.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns ``True`` if the following conditions are satisfied:
+        - layout is written in the correct form.
+        - layout matches the number of qubits in the circuit.
+    otherwise returns ``False``.
+    """
+    if not check_mapping_consistency(layout, verbose=verbose):
+        return False
+    if circuit.nqubits == len(layout):
+        if verbose:
+            log.info("Layout can be used on circuit.")
+        return True
+    elif circuit.nqubits > len(layout):
+        if verbose:
+            log.info("Layout can't be used on circuit. The circuit requires more qubits.")
+        return False
+    else:
+        if verbose:
+            log.info("Layout can't be used on circuit. Ancillary extra qubits need to be added to the circuit.")
+        return False
+
+
+def check_mapping_consistency(layout, verbose=False):
+    """Checks if layout is correct.
+
+    Args:
+        layout (dict): physical to logical qubit mapping.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns: ``True`` if layout is written in the correct form.
+    otherwise returns ``False``.
+    """
+    values = list(layout.values())
+    values.sort()
+    keys = list(layout.keys())
+    ref_keys = list("q" + str(i) for i in range(len(keys)))
```

Comment:
Can we avoid making the `"q"` mandatory?

Reply:
Now the "q" is necessary to make the code work